uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveHoursBalanceChangeCalculationTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_Service_LeaveHoursBalanceChangeCalculation as LeaveHoursBalanceChangeCalculation;
+
+/**
+ * @group headless
+ */
+class CRM_HRLeaveAndAbsences_Service_LeaveHoursBalanceChangeCalculationTest extends BaseHeadlessTest {
+
+  use CRM_HRLeaveAndAbsences_LeaveRequestHelpersTrait;
+
+  public function testGetAmountReturnsCorrectlyForFromDate() {
+    $leaveDaysCalculationService = new leaveHoursBalanceChangeCalculation();
+    $leaveRequest = $this->getLeaveRequestInstance();
+    $balanceChanges = $this->getBalanceChanges();
+    $amount = $leaveDaysCalculationService->getAmount($leaveRequest, new DateTime('2016-06-01'), $balanceChanges);
+
+    $this->assertEquals($leaveRequest->from_date_amount, $amount);
+  }
+
+  public function testGetAmountReturnsCorrectlyForToDate() {
+    $leaveDaysCalculationService = new leaveHoursBalanceChangeCalculation();
+    $leaveRequest = $this->getLeaveRequestInstance();
+    $balanceChanges = $this->getBalanceChanges();
+    $amount = $leaveDaysCalculationService->getAmount($leaveRequest, new DateTime('2016-06-04'), $balanceChanges);
```

Comment:
I think it would make things a little bit clearer if instead of `new DateTime('2016-06-04')` you did `new DateTime($leaveRequest->to_date)`. Also applies to the from_date test